float64[] destination
float64 v
float64 a
bool io_0
